redwood/web/src/pages/SignUp/ConnectWalletPage/ConnectWalletPage.tsx

```diff
@@ -1,31 +1,59 @@
+import {useLazyQuery} from '@apollo/client'
 import {Spacer, Text, VStack} from '@chakra-ui/layout'
+import {Alert, AlertIcon, AlertTitle} from '@chakra-ui/react'
 import {Redirect, routes} from '@redwoodjs/router'
 import {MetaTags} from '@redwoodjs/web'
-import {useContext} from 'react'
+import {useContext, useEffect} from 'react'
 import ConnectButton from 'src/components/ConnectButton/ConnectButton'
 import UserContext from 'src/layouts/UserContext'
+import { fetchAddressTransactions, fetchAddressTransactionsVariables } from 'types/graphql'
 import SignUpLogo from '../SignUpLogo'
 
 const ConnectWalletPage: React.FC<{
   purposeIdentifier?: string
   externalAddress?: string
 }> = () => {
   const {ethereumAddress} = useContext(UserContext)
-  if (ethereumAddress != null)
+
+  useEffect(() => {
+    if (ethereumAddress) queryAddressTransactions({variables: {ethereumAddress}})
```

Comment:
Hitting the EtherScan API on every pageload definitely feels like overkill! Don't think this should go in the `UserContext`—should live on https://github.com//zorro-project/zorro/blob/main/redwood/web/src/pages/SignUp/ConnectWalletPage/ConnectWalletPage.tsx instead, which should check that the connected address has no transactions before letting you advance through the flow.

Reply:
It is on that page, and thats what this is doing! 😄

Reply:
Ah, of course—sorry, I saw the diff for `redwood/web/src/layouts/UserContext.tsx` above (which only has one line changed) and somehow my eyes missed the fact that this was a new file 🤦.

Could you share a screenshot of what this error looks like?

Reply:
I mentioned in the Discord that I'd like to either add the ability to disconnect your wallet easily.
1. We could do it on the modal in the AccountModal
2. Or we could do it under the error message shown, making it easier for the user. Or both! It's nice to have.

![Screen Shot 2022-01-18 at 8 46 45 AM](https://user-images.githubusercontent.com/25807676/149970339-dbbf3286-d583-4be7-b218-40d2d93be62f.png)

I also mentioned that we should handle selecting multiple addresses selected on MetaMask if this was not already handled. Happy to modify the AccountModal & ConnectButton for a better experience :)

Reply:
There should be a way to bypass this warning if the user is really sure. @RoboTeddy thoughts on the copy?

Reply:
> I'd like to either add the ability to disconnect your wallet easily

I would love this! But I'm not really sure it's possible? At least calling `deactivate` in ethers.js seems to just disconnect the wallet on the current pageload, but as soon as I reload the page it gets reconnected. If you can figure out a way to reliably disconnect that seems good.

Regardless, that should probably live in another PR to keep this one focused?